### PR TITLE
Navigation

### PR DESCRIPTION
**Pull request by talsch64**

### CODE REVIEW

File: GIS.cpp [hunk 1]
```diff
@@ -262,3 +262,20 @@ bool GIS::addEntity(std::unique_ptr<Entity> entity)
...
+const Way &GIS::getWay(const EntityId &id) const {
+    Entity *entity = entities.find(id)->second.get();
+    if (entity->getType() != "Way") {
+        //TODO print to log entity is not a way
+    }
+    return *(dynamic_cast<Way*>(entity));
+}
+
+std::vector<EntityId> GIS::getWaysByJunction(const EntityId &id) const {
+    Entity *entity = entities.find(id)->second.get();
```
Comment (Royar13) [inline GIS.cpp:275]: "The function shall get EntityId of a Junction and return a vector of Ids of all the ways that start at this junction and those which end at this junction and are bidirectional"

File: Restrictions.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+
+std::unordered_set<std::string> Restrictions::split(const char *restrictions, const char* delimiter) {
+    std::unordered_set<std::string> restrictionsSet;
+    char restrictionsCopy[strlen(restrictions)];
+    strcpy(restrictionsCopy, restrictions);
+    char* restriction = strtok(restrictionsCopy, delimiter);
+    while (restriction != nullptr) {
+        restrictionsSet.insert(std::string(restriction));
+        restriction = strtok (nullptr, delimiter);
```
Comment (Royar13) [inline Restrictions.cpp:14]: Why is nullptr being split?

File: Restrictions.cpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+#include "Restrictions.h"
+
+Restrictions::Restrictions(const char *restrictions): restrictions(split(restrictions, " ,")) {}
+
+
+std::unordered_set<std::string> Restrictions::split(const char *restrictions, const char* delimiter) {
+    std::unordered_set<std::string> restrictionsSet;
+    char restrictionsCopy[strlen(restrictions)];
+    strcpy(restrictionsCopy, restrictions);
+    char* restriction = strtok(restrictionsCopy, delimiter);
```
Comment (Royar13) [inline Restrictions.cpp:11]: Aren't you assuming a space before the ","? Should probably use regex

File: entities/JsonHandlers/parsers/EntityJsonParser.cpp [hunk 4]
```diff
@@ -51,6 +51,8 @@ std::unique_ptr<Way> EntityJsonParser::parseWay(rapidjson::Value &doc, const GIS
     std::unique_ptr<Way> way = std::make_unique<Way>(id, name, description, categoryTags, std::move(geometry), from, to,
                                                      direction, speedLimit, tollRoad,
                                                      restricted);
+    ((Junction *)fromEntity)->addWay(id);
+    ((Junction *)toEntity)->addWay(id);
```
Comment (Royar13) [inline entities/JsonHandlers/parsers/EntityJsonParser.cpp:55]: We only want to add it to the end junction if it is bidirectional

**Review by Royar13 — COMMENTED** [review]
Add tests!

File: Restrictions.cpp (hunk 3 above)
Comment (talsch64) [reply Restrictions.cpp:11]: No, I intentionally added the space because it splits by "," and " "

File: Restrictions.cpp (hunk 2 above)
Comment (talsch64) [reply Restrictions.cpp:14]: This is how strtok works:
http://www.cplusplus.com/reference/cstring/strtok/

File: entities/JsonHandlers/parsers/EntityJsonParser.cpp (hunk 4 above)
Comment (talsch64) [reply entities/JsonHandlers/parsers/EntityJsonParser.cpp]: fixed

File: GIS.cpp (hunk 1 above)
Comment (talsch64) [reply GIS.cpp:281]: fixed

**Pull request merged**